js/DataService.js

```diff
@@ -8,13 +8,46 @@ angular.module('PathOfDamage')
     var int = parseInt(string);
     return isNaN(int) ? null : int;
   }
+  
+  function Row(key, value) {
+    this.enabled = true;
+    this.name = '';
+    this.value = null;
+    if (key !== undefined) {
```

Comment:
If you inspect angular objects, they all have a ton of unused properties. :P I think the rows are a consistent enough concept that including it as a property they have is perfectly acceptable. It's more readable than adding abstract properties later (especially via another constructor's input), and the actual performance impact is negligible. Also, since we only ever use that key/value thing for adding the element property, we might as well at least name it accordingly.